src/main/java/teammates/common/util/Url.java

```diff
@@ -2,14 +2,12 @@
 
 public class Url {
 
+    protected final String baseUrl;
     protected String urlString;
 
-    public Url(String url) {
-        this.urlString = url.replace(getAppUrl(), ""); // force the URL to be relative
-    }
-
-    protected String getAppUrl() {
-        return Config.APP_URL;
+    public Url(String baseUrl, String url) {
```

Comment:
There's no regulation set on stone, but with our design the baseUrl will always be trimmed (thanks to `trimTrailingSlash`) and the path will always start with `/` (thanks to the design of `Const.ActionURIs` and `Const.ViewURIs`), so we don't really need to be defensive in this way. Besides if we want to be really really defensive there's so much more to do, like checking whether the base URL starts with `http://` or `https://`, the URL is malformed, etc.